test/unit/format/vcf_input_test.cpp

```diff
@@ -0,0 +1,211 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <bio/format/vcf_input_handler.hpp>
+#include <bio/var_io/reader.hpp>
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/test/expect_range_eq.hpp>
+
+#include "vcf_data.hpp"
+
+template <bio::ownership own>
+void field_types_bcf_style()
+{
+    std::istringstream istr{std::string{example_from_spec}};
+
+    using record_t =
+      bio::detail::record_from_typelist<std::remove_cvref_t<decltype(bio::var_io::default_field_ids)>,
+                                        std::remove_cvref_t<decltype(bio::var_io::field_types_bcf_style<own>)>>;
+    ;
```

Comment:
I know this is only done by developers not users but why did you switch from 
`record<ids, types>` to `record<ids, type1, type2>`? seems more complicated from a first glance.

Reply:
At the time, it seemed to make sense, but you are right that it made other things more complicated...

Reply:
It is more tuple-like (the only difference being that IDs are prepended) and it greatly simplified the definition of `make_record()` and `tie_record()` : https://github.com/h-2/b.i.o./blob/main/include/bio/record.hpp#L342

Since these are user-facing, I think it is more important to get them right.

Reply:
I'd like to add that one also to design discussion :)

Reply:
The more I think of it, this change should be reverted.